New Prototypes and Updates to Tw2AnimationController

cppctamber opened the pull request:
- Added support for pending commands that have no arguments `_DoRebuildCachedData` @prototype
- Added `GetAnimation` @prototype which returns a loaded `Tw2Animation` by it's name, used this prototype to simplify some of the others
- Added `ResetAnimation` @prototype which resets a loaded `Tw2Animation` by it's name
- Added `PlayAnimationFrom` @prototype which plays a loaded `Tw2Animation` from a specific time. (Didn't update the existing `PlayAnimation` as it could break people's existing code)
- Updated `StopAnimation` @prototype so that it can also take a list of `Tw2Animation` names aswell as just a single name. It can now also be called before any animations have been loaded, and will stop them when they have
- Updated `StopAllAnimations` @prototype so that it can now be called before any animations have been loaded, and will stop them when they have
- Added `StopAllAnimationsExcept` @prototype that stops all `Tw2Animation`s except those that are supplied by name in an array
- Made changes to @type checking as per Filipp's suggestions
- Simplified functions a little by some changes to the `pendingCommand` @property

filipppavlov commented on src/core/Tw2AnimationController.js:
```diff
@@ -448,40 +492,138 @@ Tw2AnimationController.prototype.PlayAnimation = function(name, cycle, callback)
+        names = [names];
+    }
+
+    if (names && Object.prototype.toString.apply(names) === '[object Array]')
```
If "names" is either string or array of strings and you already handled string case, can't you just assume that names is an array at this point?

cppctamber replied:
yes.

cppctamber commented:
Changes made as per @filipppavlov 's suggestions.

cppctamber commented:
- defined `Tw2AnimationController` @property `pendingCommands` to simplify it's usage elsewhere
- removed `pendingCommands` definition in all @prototypes, they are no longer necessary
- set the default `pendingCommands` value to be an `Array` and changed the method in which pending commands were executed and reset
